src/codec/encoder/MayaEncoder.cpp

```diff
@@ -381,11 +381,14 @@ class SerializedGeometry {
 private:
 	void reserveMemory(const prtx::GeometryPtrVector& geometries,
 	               const std::vector<prtx::MaterialPtrVector>& materials) {
-		// Allocate memory for geometry
 		uint32_t numCounts = 0;
 		uint32_t numIndices = 0;
 		uint32_t maxNumUVSets = 0;
 
+		std::vector<uint32_t> numUvs(maxNumUVSets);
+		std::vector<uint32_t> numUvCounts(maxNumUVSets);
+		std::vector<uint32_t> numUvIndices(maxNumUVSets);
```

Comment:
maxNumUVSets is not properly initialized here (maxNumUVSets == 0). Removing the first loop in this way results in index out of bounds error further below. 
Please make sure numUvs, numUvCounts, numUvIndices are all properly initialized, before we access them.